lorien/Main.gd

```diff
@@ -545,3 +551,13 @@ func _on_scale_changed() -> void:
 	_canvas.set_canvas_scale(scale)
 	get_tree().set_screen_stretch(SceneTree.STRETCH_MODE_DISABLED, SceneTree.STRETCH_ASPECT_IGNORE, Vector2(0,0), scale)
 	OS.min_window_size = Config.MIN_WINDOW_SIZE * scale
+
+# --------------------------------------------------------------------------------------------------
+func _on_eraser_pressed():
+	_eraser_last_tool_active = _canvas._active_tool.get_class()
```

Comment:
Instead of doing this, you should keep an internal `Tool.Types` variable in the `InfiniteCanvas` class and expose a `get_active_tool_type()` method. Then us this here .This way you can get rid of all the `get_class()` methods and the `ToolNames `enum